sdk/src/record-scanner.ts

```diff
@@ -0,0 +1,229 @@
+        } catch (error) {
+            console.error(`Failed to get encrypted records: ${error}`);
+            throw error;
+        }
+    }
+
+    async serialNumbersExist(serialNumbers: string[]): Promise<Record<string, boolean>> {
+        try {
+            const response = await this.recordScannerServiceRequest(
+                new Request(`${this.url}/records/sns`, {
+                    method: "POST",
+                    headers: { "Content-Type": "application/json" },
+                    body: JSON.stringify(serialNumbers),
+                }),
+            );
+
+            return await response.json();
+        } catch (error) {
+            console.error(`Failed to check if serial numbers exist: ${error}`);
+            throw error;
+        }
+    }
+
+    async tagsExist(tags: string[]): Promise<Record<string, boolean>> {
+        try {
+            const response = await this.recordScannerServiceRequest(
+                new Request(`${this.url}/records/tags`, {
+                    method: "POST",
+                    headers: { "Content-Type": "application/json" },
+                    body: JSON.stringify(tags),
+                }),
+            );
+            
+            return await response.json();
+        } catch (error) {
+            console.error(`Failed to check if tags exist: ${error}`);
+            throw error;
+        }
+    }
+
+    async findRecord(searchParameters: OwnedFilter, filterFn?: (record: RecordPlaintext) => boolean): Promise<OwnedRecord> {
+        try {
+            const records = await this.findRecords(searchParameters, filterFn);
+
+            if (records.length > 0) {
+                return records[0];
+            }
+
+            throw new Error("Record not found");
+        } catch (error) {
+            console.error(`Failed to find record: ${error}`);
+            throw error;
+        }
+    }
+
+    async findRecords(searchParameters: OwnedFilter, filterFn?: (record: RecordPlaintext) => boolean): Promise<OwnedRecord[]> {
+        if (!this.uuid) {
+            throw new Error("Not registered");
+        }
+
+        const filterWithUuid: OwnedFilterWithUuid = {
+            ...searchParameters,
+            uuid: this.uuid,
+        };
+
+        try {
+            const response = await this.recordScannerServiceRequest(
+                new Request(`${this.url}/records/owned`, {
+                    method: "POST",
+                    headers: { "Content-Type": "application/json" },
+                    body: JSON.stringify(filterWithUuid),
+                }),
+            );
+
+            const records = await response.json();
+            return filterFn ? records.filter(filterFn) : records;
+        } catch (error) {
+            console.error(`Failed to get owned records: ${error}`);
+            throw error;
+        }
+    }
+
+    async findCreditsRecord(microcredits: number, searchParameters: OwnedFilter, nonces?: string[]): Promise<OwnedRecord> {
```

Comment:
```suggestion
    async findCreditsRecord(microcredits: number, searchParameters: OwnedFilter, nonces?: string[]): Promise<OwnedRecord> {
```

Can you add JSDoc documentation to each of these?

Reply:
done